pkg/skaffold/deploy/kustomize.go

```diff
@@ -205,6 +215,23 @@ func (k *KustomizeDeployer) Render(ctx context.Context, out io.Writer, builds []
 	return nil
 }
 
+// UnmarshalYAML implements JSON unmarshalling by reading an inline yaml fragment.
+func (p *patchWrapper) UnmarshalYAML(unmarshal func(interface{}) error) (err error) {
+	pp := &patchPath{}
+	_ = unmarshal(&pp)
```

Comment:
@nkubala, I'm relatively new to Golang so .... I didn't do the err check here because it will error if its a string and is part of the old API. Now that I'm thinking about it now maybe I could do something like (psuedo code)
```
err = unmarshall
if pp.path and pp.patch is nil and if err:
   err2 := unmarshal_to_string()
if err and err2:
   log error
```
Thoughts?

Reply:
i think this change is backward compatible. 
```
	if pp.Path == "" && pp.Patch == "" {
```
should imply 
``` 
if err != nil {
```

to make it explicitly 
```
pp := &patchPath{}
if err := unmarshal(&pp); err != nil{
                var oldPathString string
		if err := unmarshal(&oldPathString); err != nil {
			return err
		}

		warnings.Printf("list of file paths deprecated: see https://github.com/kubernetes-sigs/kustomize/blob/master/docs/plugins/builtins.md#patchtransformer")
		pp.Path = oldPathString
}
```

Reply:
Implemented "explicit".